src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts

```diff
@@ -0,0 +1,207 @@
+          >
+            <paper-tab
+              @click="${() => (this._selectedCard = numcards)}"
+            >
+              <ha-icon icon="hass:plus"></ha-icon>
+            </paper-tab>
+          <paper-tabs>
+        </div>
+
+        <div id="editor">
+          ${
+            selected < numcards
+              ? html`
+                  <div id="card-options">
+                    <paper-icon-button
+                      title="Move card before"
+                      icon="hass:arrow-left"
+                      ?disabled=${selected === 0}
+                      @click=${this._handleMoveBefore}
+                    ></paper-icon-button>
+
+                    <paper-icon-button
+                      title="Move card after"
+                      icon="hass:arrow-right"
+                      ?disabled=${selected === numcards - 1}
+                      @click=${this._handleMoveAfter}
+                    ></paper-icon-button>
+
+                    <paper-icon-button
+                      icon="hass:delete"
+                      @click=${this._handleDeleteCard}
+                    ></paper-icon-button>
+                  </div>
+
+                  <hui-card-editor
+                    .hass=${this.hass}
+                    .value="${this._config.cards[selected]}"
+                    @config-changed="${this._handleConfigChanged}"
+                  ></hui-card-editor>
+                `
+              : html`
+                  <hui-card-picker
+                    .hass="${this.hass}"
+                    @config-changed="${this._handleCardPicked}"
+                  ></hui-card-picker>
+                `
+          }
+        </div>
+      </div>
+    `;
+  }
+
+  private _handleConfigChanged(ev) {
+    ev.stopPropagation();
+    if (!this._config) {
+      return;
+    }
+    this._config.cards[this._selectedCard] = ev.detail.config;
+    fireEvent(this, "config-changed", { config: this._config });
+  }
+
+  private _handleCardPicked(ev) {
+    ev.stopPropagation();
+    if (!this._config) {
+      return;
+    }
+    const config = ev.detail.config;
+    this._config.cards.push(config);
+    fireEvent(this, "config-changed", { config: this._config });
+  }
+
+  private _handleDeleteCard() {
+    if (!this._config) {
+      return;
+    }
+    this._config.cards.splice(this._selectedCard, 1);
+    this._selectedCard = Math.max(0, this._selectedCard - 1);
+    fireEvent(this, "config-changed", { config: this._config });
+  }
+
+  private _handleMoveBefore() {
+    if (!this._config) {
+      return;
+    }
+    const tmp = this._config.cards[this._selectedCard - 1];
+    this._config.cards[this._selectedCard - 1] = this._config.cards[
+      this._selectedCard
+    ];
+    this._config.cards[this._selectedCard] = tmp;
+    this._selectedCard = this._selectedCard - 1;
+    fireEvent(this, "config-changed", { config: this._config });
+  }
+
+  private _handleMoveAfter() {
```

Comment:
MoveBefore and MoveAfter are the same, only the target index is different, you can do:

```ts
  private _handleMove(source: Number, target: Number) {
    if (!this._config) {
      return;
    }
    const card = this._config.cards.splice(source, 1)[0];
    this._config.cards.splice(target, 0, card);
    this._selectedCard = target;
    fireEvent(this, "config-changed", { config: this._config });
  }

  private _handleMoveAfter() {
    this._handleMove(this._selectedCard, this._selectedCard + 1);
  }
  private _handleMoveBefore() {
    this._handleMove(this._selectedCard, this._selectedCard - 1);
  }
```